src/diffusers/configuration_utils.py

```diff
@@ -154,15 +154,25 @@ def from_config(cls, pretrained_model_name_or_path: Union[str, os.PathLike], ret
 
         """
         config_dict = cls.get_config_dict(pretrained_model_name_or_path=pretrained_model_name_or_path, **kwargs)
-
         init_dict, unused_kwargs = cls.extract_init_dict(config_dict, **kwargs)
 
+        # Allow dtype to be specified on initialization
+        if "dtype" in unused_kwargs:
+            init_dict["dtype"] = unused_kwargs.pop("dtype")
+
+        # Return model and optionally state and/or unused_kwargs
         model = cls(**init_dict)
+        return_tuple = (model,)
+
+        # Some components (Flax schedulers) have a state.
+        if getattr(cls, "has_state", False):       # Check for "create_state" in model instead?
+            state = model.create_state()
+            return_tuple += (state,)
```

Comment:
We should make this a bit more robust

1. Check if it's a scheduler i.e instance of `SchedulerMixin` if yes
2. Check if the name starts with `Flax`